packages/permissions/src/components/authorized/authorized.tsx

```diff
@@ -19,19 +20,29 @@ type TDemandedActionRight = {
   group: TActionRightGroup;
   name: TActionRightName;
 };
+type TDemandedDataFence = {
+  group: string;
+  name: string;
+  type: string;
+};
 type TActionRight = {
   [key: string]: boolean;
 };
 type TActionRights = {
   [key: string]: TActionRight;
 };
-
+type TDataFence = {
+  // E.g. { store: {...} }
+  [key: string]: {};
```

Comment:
how about 
```typescript
type TDataFenceKey = "store"

type TDataFence = {
 [key: TDataFenceKey]: {};
}
```

then in the future we can take advantage of typescript unions:
`type TDataFenceType = "store" | "price" `

Reply:
that's a good idea. however, the `fetch-project` already supports passing down multiple data-fence type so we could keep this.